great_expectations/data_context/data_context.py

```diff
@@ -1141,20 +1066,60 @@ def store_validation_param(self, run_id, key, value):
         Returns:
             None
         """
-        self._evaluation_parameter_store.set(run_id, key, value)
+        run_params = self.get_parameters_in_evaluation_parameter_store_by_run_id(run_id)
+        run_params[key] = value
+        self.evaluation_parameter_store.set(run_id, run_params)
 
-    def get_validation_param(self, run_id, key):
-        """Get a new validation parameter.
+    def get_parameters_in_evaluation_parameter_store_by_run_id(self, run_id):
+        """Fetches all validation parameters for a given run_id.
 
         Args:
-            run_id: run_id for desired value
+            run_id: current run_id
             key: parameter key
+            value: parameter value
 
         Returns:
-            value stored in evaluation_parameter_store for the provided run_id and key
+            None
         """
-        return self._evaluation_parameter_store.get(run_id, key)
+        if self.evaluation_parameter_store.has_key(run_id):
+            return self.evaluation_parameter_store.get(run_id)
+        else:
+            return {}
 
+    def bind_evaluation_parameters(self, run_id):  # , expectations):
```

Comment:
I believe this comment and portion of the refactor are based on a misread of the way DataAsset uses bind evaluation parameters -- it's not using any mutability of the bind parameters.

Reply:
Yes, this was a misread.